### ISSUE TITLE
Infinite loop protection

### ISSUE THREAD

**Issue by sjoerdsein** [markusfisch/ShaderEditor#47]
This app appears to have no protection against infinite loops. My device crashed when I accidentally typed:
```c++
for (int i; i < x; i = i) { // Was to become i = i + 1
  ...
}
```
It crashed because it ran the code before I was able to type ` + 1`. Luckily it didn't autosave...

I'm using a OnePlus One with Android 6.0.1 and Cyanogen OS 13.1.2-ZNH2KAS3P0-bacon. App version is 2.4.1.

PS: I know `i++` is a thing but I was just experimenting.

**Comment by rafaellago** [comment]
Some protection for infinite loop would be nice, but to help with your experimentation. try using `i += n` 

Should be enough to avoid problems. =)

**Comment by markusfisch** [comment]
Good point. Although I think there's no way to actually stop a running (shader) program, we might get some warning from [glValidateProgram](https://www.opengl.org/sdk/docs/man/html/glValidateProgram.xhtml).
